client/extensions/woocommerce/state/selectors/promotions.js

```diff
@@ -1,26 +1,60 @@
 /**
+ * External dependencies
+ */
+import { get, find, isObject } from 'lodash';
+
+/**
  * Internal dependencies
  */
 import { getSelectedSiteWithFallback } from '../sites/selectors';
 
 export function getPromotions( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
-	const { promotions } = rootState.extensions.woocommerce.sites[ siteId ];
+	const promotions = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId, 'promotions' ], {} );
 	return promotions.promotions;
 }
 
+export function getPromotion( rootState, promotionId, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const promotions = getPromotions( rootState, siteId );
+	return ( find( promotions, ( p ) => promotionId === p.id ) || null );
```

Comment:
Suggested: lodash find instead

Reply:
This is lodash `find` -- see import above.